test/model/test_store.dart

```diff
@@ -166,6 +177,59 @@ class TestGlobalStore extends GlobalStore {
   }
 }
 
+/// A [GlobalStore] that causes no database queries,
+/// and loads per-account data from API responses prepared by callers.
+///
+/// The per-account stores will use [FakeApiConnection].
+///
+/// Like [LiveGlobalStore] and unlike [TestGlobalStore],
+/// account data is loaded via [UpdateMachine.load].
+/// Callers can set [prepareRegisterQueueResponse]
+/// to prepare a register-queue payload or an exception.
+/// The implementation pauses the event-polling loop
+/// to avoid being a nuisance and does a boring
+/// [FakeApiConnection.prepare] for the register-token request.
+///
+/// See also:
+///   * [TestGlobalStore], which prepares per-account data
+///     without using [UpdateMachine.load].
+class UpdateMachineTestGlobalStore extends GlobalStore with _ApiConnectionsMixin, _DatabaseMixin {
```

Comment:
I spent some time staring at this vs. TestGlobalStore and thinking about the relationship between them. I didn't see any big changes I'd want to make; but the `updateMachines` field over on that other class was one thing that was a bit incongruous with the naming of the classes, and I realized it wasn't needed.

I'll push an extra commit that cuts that out, as well as a prep commit in a different area that this brought up:
dd9b29bb8 test [nfc]: Simplify out TestGlobalStore.updateMachines
247bf7ffe store test [nfc]: Clarify what prepareReload is doing

Reply:
Great; thanks!